File: packages/sdk/typescript/human-protocol-sdk/src/graphql/queries/reward.ts
import gql from 'graphql-tag';

const REWARD_ADDED_EVENT_FRAGMENT = gql`
  fragment RewardAddedEventFields on RewardAddedEvent {
    escrowAddress
    staker
    slasher
    amount
  }
`;

export const GET_REWARD_ADDED_EVENTS_QUERY = gql`
  query GetRewardAddedEvents($slasherAddress: String!) {
    rewardAddedEvents(where: { slasher: $slasherAddress }) {
      ...EscrowFields
    }
  }
  ${REWARD_ADDED_EVENT_FRAGMENT}
`;

File: packages/sdk/python/human-protocol-sdk/human_protocol_sdk/gql/reward.py
reward_added_event_fragment = """
fragment RewardAddedEventFields on RewardAddedEvent {
    escrowAddress
    staker
    slasher
    amount
}
"""

get_reward_added_events_query = """
query GetRewardAddedEvents($slasherAddress: String!) {{
    rewardAddedEvents(where: {{ slasher: $slasherAddress }}) {{
        ...RewardAddedEventFields
    }}
}}
{reward_added_event_fragment}
""".format(
    reward_added_event_fragment=reward_added_event_fragment
)
